File: src/main/java/com/fx/billsoft/domain/billnote/service/BillService.java
package com.fx.billsoft.domain.billnote.service;

import com.fx.billsoft.domain.billnote.entity.Bill;
import com.fx.billsoft.infrastructure.billnote.repository.BillRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

@Service
public class BillService {
    private final BillRepository billRepository;

    @Autowired
    public BillService(BillRepository billRepository) {
        this.billRepository = billRepository;
    }

    public Bill createBill(Bill bill) {
        return billRepository.save(bill);
    }

    public Optional<Bill> getBillById(Long id) {
        return billRepository.findById(id);
    }

    public Bill updateBill(Bill bill) {
        return billRepository.save(bill);
    }

    public void deleteBill(Long id) {
        billRepository.deleteById(id);
    }

    public List<Bill> getBillsByLedgerId(Long ledgerId) {
        return billRepository.findByLedgerId(ledgerId);
    }

    public List<Bill> getBillsByUserId(Long userId) {
        return billRepository.findByUserId(userId);
    }

    public List<Bill> getBillsByLedgerIdAndUserId(Long ledgerId, Long userId) {
        return billRepository.findByLedgerIdAndUserId(ledgerId, userId);
    }
}

File: frontend/src/services/billService.js
import http from './http'

class BillService {
  // 获取所有账单（分页）
  getBills(page = 0, size = 10) {
    return http.get(`/bills?page=${page}&size=${size}&sort=date,desc`)
  }

  // 根据用户ID获取账单（分页）
  getBillsByUserId(userId, page = 0, size = 10) {
    return http.get(`/bills/user/${userId}?page=${page}&size=${size}&sort=date,desc`)
  }

  // 根据账本ID获取账单（分页）
  getBillsByLedgerId(ledgerId, page = 0, size = 10) {
    return http.get(`/bills/ledger/${ledgerId}?page=${page}&size=${size}&sort=date,desc`)
  }

  // 根据账本ID和用户ID获取账单（分页）
  getBillsByLedgerIdAndUserId(ledgerId, userId, page = 0, size = 10) {
    return http.get(`/bills/ledger/${ledgerId}/user/${userId}?page=${page}&size=${size}&sort=date,desc`)
  }

  // 获取账单详情
  getBill(id) {
    return http.get(`/bills/${id}`)
  }

  // 创建账单
  createBill(bill) {
    return http.post('/bills', bill)
  }

  // 更新账单
  updateBill(id, bill) {
    return http.put(`/bills/${id}`, bill)
  }

  // 删除账单
  deleteBill(id) {
    return http.delete(`/bills/${id}`)
  }
}

export default new BillService()